File: IDATT2106 Systemutvikling 2 med smidig prosjekt/backend/src/main/java/com/ntnu/gidd/dto/User/UserListDto.java
package com.ntnu.gidd.dto.User;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserListDto extends ContextAwareUserDto {
    private UUID id;
    private String firstName;
    private String surname;
    private String email;
    private String image;
}

File: IDATT2105 Full-stack applikasjonsutvikling/Rombestilling/backend/src/main/kotlin/ntnu/idatt2105/user/dto/UserListDto.kt
package ntnu.idatt2105.user.dto

import ntnu.idatt2105.user.model.User
import java.util.*

data class UserListDto(
    val id: UUID? = null,
    val firstName: String = "",
    val surname: String = "",
    val phoneNumber: String = "",
    val email: String = ""
)
fun User.toUserListDto() = UserListDto(
        id = this.id,
        firstName = this.firstName,
        surname = this.surname,
        email = this.email,
        phoneNumber = this.phoneNumber

)
